src/pages/Swap/index.tsx

```diff
@@ -123,11 +122,11 @@ const DetailsSwapSection = styled(SwapSection)`
 `
 
 function getIsValidSwapQuote(
-  trade: InterfaceTrade<Currency, Currency, TradeType> | undefined,
+  trade: InterfaceTrade | undefined,
   tradeState: TradeState,
   swapInputError?: ReactNode
 ): boolean {
-  return !!swapInputError && !!trade && (tradeState === TradeState.VALID || tradeState === TradeState.SYNCING)
+  return Boolean(swapInputError && trade && tradeState === TradeState.VALID)
```

Comment:
This is confusing (and it was before too) - "isValidSwapQuote" will return false when swapInputError is null, right? Is this code correct?